### PR TITLE
Add event stitching support

### PR DESCRIPTION
**Pull request by kevinlind**
## Description

Add event stitching support by using `Event.chainEventWithName` API when creating new events. 

Adds some SwiftLint fixes:

- Use correct .swiftlint.yml configuration with unit and functional tests
- Set execute permission to pre-commit GitHook to run SwiftLint on commits
- Fix SwiftLint warnings

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: Sources/EdgeBridge.swift [hunk 1]
```diff
@@ -55,7 +55,7 @@ public class EdgeBridge: NSObject, Extension {
             return
         }
 
-        dispatchTrackRequest(data: eventData, timestamp: event.timestamp)
+        dispatchTrackRequest(data: eventData, timestamp: event.timestamp, parentEvent: event)
```
[the remark this answers is not in the document]
Comment (addb) [reply Sources/EdgeBridge.swift]: That we could even extract the eventData from the event.

[the remark this answers is not in the document]
Comment (kevinlind) [reply Sources/EdgeBridge.swift]: For events from Rules, the data is the consequence detail and not the Event data, so the `data` parameter is still needed.

File: Sources/EdgeBridge.swift [hunk 2]
```diff
@@ -83,28 +83,28 @@ public class EdgeBridge: NSObject, Extension {
...
-        dispatchTrackRequest(data: consequenceDetail, timestamp: event.timestamp)
+        dispatchTrackRequest(data: consequenceDetail, parentEvent: event)
     }
 
     /// Helper to create and dispatch an experience event.
     /// - Parameters:
     ///   - data: dictionary containing free-form data to send to Edge Network
-    ///   - timestamp: timestamp of Event
-    private func dispatchTrackRequest(data: [String: Any], timestamp: Date) {
+    ///   - parentEvent: the triggering parent event
```
Comment (emdobrin) [inline Sources/EdgeBridge.swift:92]: can we update description here, e.g. 
parentEvent: the triggering parent event used for event chaining; its timestamp is set as xdm.timestamp

File: Sources/EdgeBridge.swift [hunk 3]
```diff
@@ -83,28 +83,28 @@ public class EdgeBridge: NSObject, Extension {
...
         ]
 
-        let xdmEvent = Event(name: EdgeBridgeConstants.EventNames.EDGE_BRIDGE_REQUEST,
-                             type: EventType.edge,
-                             source: EventSource.requestContent,
-                             data: xdmEventData)
+        let event = parentEvent.createChainedEvent(name: EdgeBridgeConstants.EventNames.EDGE_BRIDGE_REQUEST,
+                                                   type: EventType.edge,
+                                                   source: EventSource.requestContent,
+                                                   data: xdmEventData)
```
Comment (emdobrin) [inline Sources/EdgeBridge.swift:105]: nit: formatting seems off here

**Review by emdobrin — APPROVED** [review]

**Pull request merged**